### PR TITLE
Modifications to WW3 for drag coefficient in the Med Sea

### PR DESCRIPTION
**Pull request by francescococetta**
# Pull Request Summary
 

### Check list  

- [ ] Branch is up to date with the authoritative repository (NOAA-EMC) develop branch. 
- [ ] Checked the [checklist for a developer submitting to develop](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-a-developer-submitting-to-develop). 
- [ ] If a version number update is required, checked the [updating version number](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-updating-version-number) checklist. 
- [ ] If a new feature was added, a regression test for testing the new feature is added. 
 
### Testing

* How were these changes tested?
* Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)
* Have the matrix regression tests been run (if yes, please note HPC and compiler)?
* Please indicate the expected changes in the regression test output, (Note the [list](https://github.com/NOAA-EMC/WW3/wiki/How-to-use-matrix.comp-to-compare-regtests-with-develop#4-look-at-results) of known non-identical tests.)
* Please provide the summary output of matrix.comp (_matrix.Diff.txt_, _matrixCompFull.txt_ and _matrixCompSummary.txt_):

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,4 @@
+###### FC
```
Comment (daniele-peano) [inline README.md]: you can remove this now

File: model/src/constants.F90 [hunk 2]
```diff
@@ -69,7 +69,9 @@ MODULE CONSTANTS
 !mdo  *** Changing nu_water to be consistent with DWAT=1000 (assumes 10degC)
       REAL, PARAMETER         :: nu_water  = 1.31E-6    !mdo   WAS: 3.E-6        
       REAL, PARAMETER         :: sed_sg  = 2.65        
-      REAL, PARAMETER         :: KAPPA = 0.40       !Von Karman's constant
+!      REAL, PARAMETER         :: KAPPA = 0.40       !Von Karman's constant
```
Comment (daniele-peano) [inline model/src/constants.F90]: you can remove these two commented lines

File: model/src/w3iogomd.F90 [hunk 3]
```diff
@@ -1071,12 +1071,17 @@ SUBROUTINE W3FLDTOIJ(FLD, I, J, IAPROC, NAPOUT, NDSEN)
...
+        I = 10
+        J = 1
+#endif
       CASE('U1')
         I = 10
         J = 1
       CASE('U2')
         I = 10
-        J = 1
+        J = 2
```
Comment (daniele-peano) [inline model/src/w3iogomd.F90:1084]: do you want to change this value?

File: model/src/w3ounfmetamd.F90 [hunk 4]
```diff
@@ -4032,14 +4032,14 @@ SUBROUTINE DEFAULT_META()
 ! IFI=10, IFJ=1
       META => GROUP(10)%FIELD(1)%META
```
Comment (daniele-peano) [inline model/src/w3ounfmetamd.F90:4033]: You could keep a group 11 for the user_defined case.

File: model/src/w3src3md.F90 [hunk 5]
```diff
@@ -65,13 +65,16 @@ MODULE W3SRC3MD
 !/ Public variables
 !/
       !air kinematic viscosity (used in WAM)
-      INTEGER, PARAMETER      :: ITAUMAX=200,JUMAX=200
-      INTEGER, PARAMETER      :: IUSTAR=100,IALPHA=200, ILEVTAIL=50
+!      INTEGER, PARAMETER      :: ITAUMAX=200,JUMAX=200
+!      INTEGER, PARAMETER      :: IUSTAR=100,IALPHA=200, ILEVTAIL=50
```
Comment (daniele-peano) [inline model/src/w3src3md.F90]: Remove these commented lines

File: model/src/w3src3md.F90 [hunk 6]
```diff
@@ -65,13 +65,16 @@ MODULE W3SRC3MD
...
+      INTEGER, PARAMETER      :: ITAUMAX=100,JUMAX=100
+      INTEGER, PARAMETER      :: IUSTAR=100,IALPHA=100, ILEVTAIL=50
       REAL                    :: TAUT(0:ITAUMAX,0:JUMAX), DELTAUW, DELU
       ! Table for H.F. stress as a function of 2 variables
       REAL                    :: TAUHFT(0:IUSTAR,0:IALPHA), DELUST, DELALP
       REAL,    PARAMETER      :: UMAX    = 50.
-      REAL,    PARAMETER      :: TAUWMAX = 2.2361 !SQRT(5.)
+!      REAL,    PARAMETER      :: TAUWMAX = 2.2361 !SQRT(5.)
```
Comment (daniele-peano) [inline model/src/w3src3md.F90]: remove this commented line

File: model/src/w3src3md.F90 [hunk 7]
```diff
@@ -464,15 +467,15 @@ SUBROUTINE W3SIN3 (A, CG, K, U, USTAR, DRAT, AS, USDIR, Z0, CD,    &
 ! I got it from, maybe just made up from drag law ... 
 !
 #ifdef W3_STAB3
-      Usigma=MAX(0.,-0.025*AS)
-      USTARsigma=(1.0+U/(10.+U))*Usigma
+      !Usigma=MAX(0.,-0.025*AS)
```
Comment (daniele-peano) [inline model/src/w3src3md.F90]: you can delete these lines if you don't need them.

File: model/src/w3src3md.F90 [hunk 8]
```diff
@@ -464,15 +467,15 @@ SUBROUTINE W3SIN3 (A, CG, K, U, USTAR, DRAT, AS, USDIR, Z0, CD,    &
...
+      !USTARsigma=(1.0+U/(10.+U))*Usigma
 #endif
       UST=USTAR
       ISTAB=3
 #ifdef W3_STAB3
-      DO ISTAB=1,2
-      IF (ISTAB.EQ.1) UST=USTAR*(1.-USTARsigma)
-      IF (ISTAB.EQ.2) UST=USTAR*(1.+USTARsigma)
+      !DO ISTAB=1,2
+      !IF (ISTAB.EQ.1) UST=USTAR*(1.-USTARsigma)
```
Comment (daniele-peano) [inline model/src/w3src3md.F90]: you can delete these lines if you don't need them

File: model/src/w3src3md.F90 [hunk 9]
```diff
@@ -549,12 +552,12 @@ SUBROUTINE W3SIN3 (A, CG, K, U, USTAR, DRAT, AS, USDIR, Z0, CD,    &
...
-      END DO 
-      D(:)=0.5*(DSTAB(1,:)+DSTAB(2,:))
-      XSTRESS=0.5*(STRESSSTAB(1,1)+STRESSSTAB(2,1))
-      YSTRESS=0.5*(STRESSSTAB(1,2)+STRESSSTAB(2,2))
-      TAUWNX=0.5*(STRESSSTABN(1,1)+STRESSSTABN(2,1))
-      TAUWNY=0.5*(STRESSSTABN(1,2)+STRESSSTABN(2,2))
+      !END DO 
+      !D(:)=0.5*(DSTAB(1,:)+DSTAB(2,:))
+      !XSTRESS=0.5*(STRESSSTAB(1,1)+STRESSSTAB(2,1))
+      !YSTRESS=0.5*(STRESSSTAB(1,2)+STRESSSTAB(2,2))
```
Comment (daniele-peano) [inline model/src/w3src3md.F90]: you can delete these lines if you don't need them

File: model/src/w3src3md.F90 [hunk 10]
```diff
@@ -595,7 +598,7 @@ SUBROUTINE W3SIN3 (A, CG, K, U, USTAR, DRAT, AS, USDIR, Z0, CD,    &
       IND  = MAX(1,MIN (IUSTAR-1, INT(XI)))
       DELI1= MAX(MIN (1. ,XI-FLOAT(IND)),0.)
       DELI2= 1. - DELI1
-      XJ=MAX(0.,(GRAV*Z0/MAX(UST,0.00001)**2-AALPHA) / DELALP)
+      XJ=MAX(0.,(GRAV*Z0/MAX(UST,0.00001)**2-AALPHA) / DELALP) ! AIMIE already MAX(UST,0.00001)
```
Comment (daniele-peano) [inline model/src/w3src3md.F90]: you can remove the comment

File: model/src/w3src3md.F90 [hunk 11]
```diff
@@ -927,7 +930,8 @@ SUBROUTINE TABU_TAUHF(FRMAX)
       REAL                    :: USTARM, ALPHAM
       REAL                    :: CONST1, OMEGA, OMEGAC 
       REAL                    :: UST, ZZ0,OMEGACC, CM
-      INTEGER, PARAMETER      :: JTOT=250
+      !INTEGER, PARAMETER      :: JTOT=250
```
Comment (daniele-peano) [inline model/src/w3src3md.F90]: you can remove this commented line

File: model/src/w3src3md.F90 [hunk 12]
```diff
@@ -942,7 +946,8 @@ SUBROUTINE TABU_TAUHF(FRMAX)
 #endif
 !
       USTARM = 5.
-      ALPHAM = 20.*AALPHA
+      !ALPHAM = 20.*AALPHA
```
Comment (daniele-peano) [inline model/src/w3src3md.F90]: you can remove this commented line

File: model/src/w3src3md.F90 [hunk 13]
```diff
@@ -960,7 +965,7 @@ SUBROUTINE TABU_TAUHF(FRMAX)
          DO K=0,IUSTAR
             UST      = MAX(REAL(K)*DELUST,0.000001)
             ZZ00       = UST**2*AALPHA/GRAV
-            IF (ZZ0MAX.NE.0) ZZ00=MIN(ZZ00,ZZ0MAX)
+            !IF (ZZ0MAX.NE.0) ZZ00=MIN(ZZ00,ZZ0MAX)
```
Comment (daniele-peano) [inline model/src/w3src3md.F90]: you can remove this line

File: model/src/w3src3md.F90 [hunk 14]
```diff
@@ -980,7 +985,8 @@ SUBROUTINE TABU_TAUHF(FRMAX)
                ZLOG     = MIN(ALOG(ZMU),0.)
                ZBETA        = CONST1*ZMU*ZLOG**4
                ! Power of Y in denominator should be FACHFE-4
-               TAUHFT(K,L)  = TAUHFT(K,L)+W(J)*ZBETA/Y*DELY
+               !TAUHFT(K,L)  = TAUHFT(K,L)+W(J)*ZBETA/Y*DELY
```
Comment (daniele-peano) [inline model/src/w3src3md.F90]: you can remove this commented line

File: model/src/w3src3md.F90 [hunk 15]
```diff
@@ -1081,7 +1087,7 @@ SUBROUTINE CALC_USTAR(WINDSPEED,TAUW,USTAR,Z0,CHARN)
       DELI2   = 1. - DELI1
       XJ      = WINDSPEED/DELU
       J       = MIN ( JUMAX-1, INT(XJ) )
-      DELJ1   = MIN(1.,XJ - REAL(J))
+      DELJ1   = MIN(1.,XJ - REAL(J))    !AIMIE alreydy corrected
```
Comment (daniele-peano) [inline model/src/w3src3md.F90]: you can remove the comment

**Review by daniele-peano — CHANGES_REQUESTED** [review]
I put some comments for cleaning the code

File: model/src/w3iogomd.F90 (hunk 3 above)
Comment (francescococetta) [reply model/src/w3iogomd.F90:1084]: yes, when 'U2' J=2

File: model/src/w3ounfmetamd.F90 (hunk 4 above)
Comment (francescococetta) [reply model/src/w3ounfmetamd.F90:4033]: done!

File: model/src/w3src3md.F90 [hunk 16]
```diff
@@ -552,12 +540,6 @@ SUBROUTINE W3SIN3 (A, CG, K, U, USTAR, DRAT, AS, USDIR, Z0, CD,    &
         TAUWNY =STRESSSTABN(3,2)
 !             WRITE(995,'(A,11G14.5)') 'NEGSTRESS:    ',TAUWNX,TAUWNY,FW*UORB**3
 #ifdef W3_STAB3
```
Comment (daniele-peano) [inline model/src/w3src3md.F90]: Remove also these #ifdef -> #endif lines

**Review by daniele-peano — CHANGES_REQUESTED** [review]

**Review by daniele-peano — APPROVED** [review]

**Pull request merged**